pkg/client/example_test.go

```diff
@@ -0,0 +1,179 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client_test
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	c client.Client
+)
+
+// This example shows how to use the client with typed and unstrucurted objects to retrieve a objects.
+func ExampleClient_get() {
+	// Using a typed object.
+	p := &corev1.Pod{}
+	// c is a created client.
```

Comment:
I was wondering if we should have an example of constructing the client as well. Because sometime people wants to create a client without using the kubebuilder or manager machinery and they get stuck. Or we can probably have another example to create a client ? something that this gist captures --> https://gist.github.com/droot/3a60b97c8b48f8d470796302e87a01ea